### PR TITLE
Add ResNeSt

### PR DESCRIPTION
**Pull request by xvjiarui**

### CODE REVIEW

**Comment by yl-1993** [comment]
Overall it looks good to me. Please kindly sign the license agreement and add the converted pretrained models. Thanks.

**Comment by xvjiarui** [comment]
Hi @yl-1993 
The model links are as followed, [ResNeSt50](https://openmmlab.oss-cn-hangzhou.aliyuncs.com/mmclassification/v0/imagenet/resnest50_converted-1ebf0afe.pth): top1_acc: 81.13, top5_acc: 95.59, [ResNeSt101](https://openmmlab.oss-cn-hangzhou.aliyuncs.com/mmclassification/v0/imagenet/resnest101_converted-032caa52.pth): top1_acc: 82.32, top5_acc: 96.24.

File: mmcls/models/backbones/resnest.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+from ..builder import BACKBONES
+from .resnet import Bottleneck as _Bottleneck
+from .resnet import ResLayer, ResNetV1d
+
+
+class RSoftmax(nn.Module):
+    """Radix Softmax module in ``SplitAttentionConv2d``.
+
+        radix (int): Radix of input.
```
Comment (hellock) [inline mmcls/models/backbones/resnest.py:15]: Args

File: docs/model_zoo.md [hunk 2]
```diff
@@ -12,6 +12,8 @@ The ResNet family models below are trained by standard data augmentations, i.e.,
...
 | ResNet-50             | 25.56     | 4.12     | 76.55 | 93.15 | [model](https://openmmlab.oss-accelerate.aliyuncs.com/mmclassification/v0/imagenet/resnet50_batch256_20200708-cfb998bf.pth) &#124; [log](https://openmmlab.oss-accelerate.aliyuncs.com/mmclassification/v0/imagenet/resnet50_batch256_20200708-cfb998bf.log.json) |
 | ResNet-101            | 44.55     | 7.85     | 78.18 | 94.03 | [model](https://openmmlab.oss-accelerate.aliyuncs.com/mmclassification/v0/imagenet/resnet101_batch256_20200708-753f3608.pth) &#124; [log](https://openmmlab.oss-accelerate.aliyuncs.com/mmclassification/v0/imagenet/resnet101_batch256_20200708-753f3608.log.json) |
+| ResNeSt-50            | 27.48     | 5.41     | 81.13 | 95.59 | [model](https://openmmlab.oss-cn-hangzhou.aliyuncs.com/mmclassification/v0/imagenet/resnest50_converted-1ebf0afe.pth) &#124; [log]() |
+| ResNeSt-101           | 48.28     | 10.27     | 82.32 | 96.24 | [model](https://openmmlab.oss-cn-hangzhou.aliyuncs.com/mmclassification/v0/imagenet/resnest101_converted-032caa52.pth) &#124; [log]() |
```
Comment (hellock) [inline docs/model_zoo.md:16]: How about 152 and 200?

Comment (xvjiarui) [reply docs/model_zoo.md]: ResNeSt only has 200, 260.

**Review by hellock — APPROVED** [review]

**Comment by hu5tao** [comment]
hello,how to converted official model into mmcls ResNest pretrained model？ for help，thanks a lot

**Comment by Ezra-Yu** [comment]
@hu5tao you can refer to the [efficientnet_to_mmcls.py](https://github.com/open-mmlab/mmclassification/blob/master/tools/convert_models/efficientnet_to_mmcls.py) script.

The weights of PyTorch are stored using ordereddict, which is a `dict` data structure. The weight difference between mmcls and the original repo.  are the names of the keys. Finding the corresponding weight and modify the names of the keys would be OK.

**Comment by hu5tao** [comment]
OK&nbsp; thank you! i get it




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "open-mmlab/mmclassification"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年4月21日(星期四) 下午4:17
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [open-mmlab/mmclassification] Add ResNeSt (#25)





 
@hu5tao you can refer to the efficientnet_to_mmcls.py script.
 
The weights of PyTorch are stored using ordereddict, which is a dict data structure. The weight difference between mmcls and the original repo.  are the names of the keys. Finding the corresponding weight and modify the names of the keys would be OK.

**Pull request merged**